Mathlib/Topology/Algebra/Module/WeakDual.lean

```diff
@@ -326,6 +353,64 @@ theorem coe_map (f : E →L[𝕜] F) : (WeakSpace.map f : E → F) = f :=
   rfl
 #align weak_space.coe_map WeakSpace.coe_map
 
+/-- The canonical preimage in `E` of an open set in `WeakSpace 𝕜 E` is open in `E`. -/
+theorem isOpen_of_isOpen_WeakSpace (U : Set (WeakSpace 𝕜 E))
+    (hU : IsOpen[(WeakSpace.instTopologicalSpace : TopologicalSpace (WeakSpace 𝕜 E))] U) :
+    IsOpen ((continuousLinearMapToWeakSpace 𝕜 E) ⁻¹' U) := by
+  exact ((continuousLinearMapToWeakSpace 𝕜 E).cont).isOpen_preimage U hU
+
+/-- A set in `E` which is open in the weak topology is open. -/
+theorem isOpen_of_isOpen_WeakSpace' (V : Set E)
+    (hV : IsOpen[(WeakSpace.instTopologicalSpace)]
+    ((continuousLinearMapToWeakSpace 𝕜 E) '' V : Set (WeakSpace 𝕜 E))) : IsOpen V := by
```

Comment:
```suggestion
/-- The canonical map `E →L[𝕜] WeakSpace 𝕜 E` is open. -/
theorem isOpenMap (continuousLinearMapToWeakSpace 𝕜 E) := by
  intro V hV
```

Reply:
Do you mean something like
`theorem isOpenMap_continuousLinearMapToWeakSpace : IsOpenMap (continuousLinearMapToWeakSpace 𝕜 E)`? But this is not true...

Or do you mean
`theorem isOpenMap_inv_toWeakSpace : IsOpenMap (ToWeakSpace 𝕜 E).symm`? I included this.